test/models/detection_test.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: detections
+#
+#  id               :integer          not null, primary key
+#  term_id          :integer          not null
+#  detector_id      :integer          not null
+#  detector_version :string
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#
+require 'test_helper'
+
+class DetectionTest < ActiveSupport::TestCase
+  test 'duplicate detections are not allowed' do
+    initial_count = Detection.count
+
+    sample = {
+      term: terms('hi'),
+      detector: detectors('doi')
+    }
+
+    Detection.create!(sample)
+
+    post_count = Detection.count
+
+    assert_equal(initial_count + 1, post_count)
+
+    assert_raises(ActiveRecord::RecordNotUnique) do
+      Detection.create!(sample)
+    end
+
+    post_duplicate_count = Detection.count
+
+    assert_equal(post_count, post_duplicate_count)
+  end
+
+  test 'new detections are allowed when detector_version is updated' do
+    initial_count = Detection.count
+
+    sample = Detection.first
+
+    new_sample = {
+      term: sample.term,
+      detector: sample.detector
+    }
+
+    # A purely duplicate record fails to save...
+    assert_raises(ActiveRecord::RecordNotUnique) do
+      Detection.create!(new_sample)
+    end
+
+    # ...but when we update the DETECTOR_VERSION env, now the same record does save.
+    new_version = 'updated'
+
+    assert_not_equal(ENV.fetch('DETECTOR_VERSION'), new_version)
+
+    ENV['DETECTOR_VERSION'] = new_version
```

Comment:
Setting env in line like this may lead to instability in the test suite. We often use the ClimateControl gem to help mitigate those risks in our tests. I'm not sure if we've used that yet in this application, but if we need to change env in the test suite I think it ensures it gets set back after the test in a safer way.